internal/persistence/telemetry_test.go

```diff
@@ -0,0 +1,243 @@
+						},
+						nil,
+					)
+			},
+			params: userID,
+			want: &datatypes.User{
+				UUID:      userID,
+				CreatedAt: createdAt,
+				Name:      ref.New("Test User"),
+			},
+		},
+		{
+			name: "Errors when reading a user",
+			mock: func(svc *fake.MockService) {
+				svc.EXPECT().
+					ReadUser(mockutil.Context(), userID).
+					Return(nil, errors.New("not found error: db: user not found"))
+			},
+			params:  userID,
+			wantErr: "ReadUser: not found error: db: user not found",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			svc := fake.NewMockService(gomock.NewController(t))
+			if tt.mock != nil {
+				tt.mock(svc)
+			}
+
+			tracer := noop.NewTracerProvider().Tracer("test")
+			w := persistence.WithTelemetry(svc, tracer)
+
+			got, err := w.ReadUser(t.Context(), tt.params)
+			if tt.wantErr != "" {
+				assert.Error(t, err, tt.wantErr)
+				return
+			}
+
+			assert.NilError(t, err)
+			assert.DeepEqual(t, got, tt.want)
+		})
+	}
+}
+
+func TestReadJWTUser(t *testing.T) {
+	t.Parallel()
+
+	userID := uuid.New()
+	createdAt := ref.New(time.Now().Truncate(time.Second))
+
+	type params struct {
+		iss string
+		sub string
+	}
+
+	type test struct {
+		name    string
+		mock    func(*fake.MockService)
+		args    params
+		want    *datatypes.User
+		wantErr string
+	}
+	for _, tt := range []test{
+		{
+			name: "Reads a user using JWT data",
+			mock: func(svc *fake.MockService) {
+				svc.EXPECT().
+					ReadJWTUser(
+						mockutil.Context(),
+						"https://oidc.example.com",
+						"1234567890",
+					).
+					Return(
+						&datatypes.User{
+							UUID:      userID,
+							CreatedAt: createdAt,
+							Name:      ref.New("Test User"),
+							JWTIss:    ref.New("https://oidc.example.com"),
+							JWTSub:    ref.New("1234567890"),
+						},
+						nil,
+					)
+			},
+			args: params{iss: "https://oidc.example.com", sub: "1234567890"},
+			want: &datatypes.User{
+				UUID:      userID,
+				CreatedAt: createdAt,
+				Name:      ref.New("Test User"),
+				JWTIss:    ref.New("https://oidc.example.com"),
+				JWTSub:    ref.New("1234567890"),
+			},
+		},
+		{
+			name: "Errors when reading a user using JWT data",
+			mock: func(svc *fake.MockService) {
+				svc.EXPECT().
+					ReadJWTUser(mockutil.Context(), "", "").
+					Return(nil, errors.New("not found error: db: user not found"))
+			},
+			args:    params{iss: "", sub: ""},
+			wantErr: "ReadJWTUser: not found error: db: user not found",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			svc := fake.NewMockService(gomock.NewController(t))
+			if tt.mock != nil {
+				tt.mock(svc)
+			}
+
+			tracer := noop.NewTracerProvider().Tracer("test")
+			w := persistence.WithTelemetry(svc, tracer)
+
+			got, err := w.ReadJWTUser(t.Context(), tt.args.iss, tt.args.sub)
+			if tt.wantErr != "" {
+				assert.Error(t, err, tt.wantErr)
+				return
+			}
+
+			assert.NilError(t, err)
+			assert.DeepEqual(t, got, tt.want)
+		})
+	}
+}
```

Comment:
It's nice to get some coverage for these telemetry functions, thanks! Ideally, we would test the span creation, etc. instead of using a noop tracer provider, but that can happen in another PR.